### PR TITLE
Documentation artifact are also uploaded as GitHub Actions Artifacts

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Hey - I updated doc building step to upload generated documentation as an artifact - this way you can very easily open the .zip file and see documentation generated by all PRs.

**Comment by potiuk** [comment]
Ready to merge :)

File: scripts/ci/libraries/_initialization.sh [hunk 1]
```diff
@@ -148,6 +151,23 @@ function initialize_common_environment {
...
+        "--env" "VERBOSE_COMMANDS"
+        "--env" "HOST_USER_ID"
+        "--env" "HOST_GROUP_ID"
+        "--env" "HOST_OS"
+        "--env" "HOST_HOME"
+        "--env" "HOST_AIRFLOW_SOURCES"
+        "--env" "PYTHON_MAJOR_MINOR_VERSION"
+        "--env" "VERSION_SUFFIX_FOR_PYPI"
+        "--env" "VERSION_SUFFIX_FOR_SVN"
+        "--env" "CI"
```
Comment (feluelle) [inline scripts/ci/libraries/_initialization.sh]: How about using [--env-file](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) ?

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh]: I was under the impression that env-file requires var=value but I looked it up and we could indeed change it this way, Let me see if I can make it nicely :)

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh]: I did it and made it even nicer :).

**Comment by potiuk** [comment]
All Green!

**Review by turbaszek — APPROVED** [review]
LGTM! But WOW... I didn't know that each build generates 1GB+ of artifacts

File: scripts/ci/libraries/_initialization.sh [hunk 2]
```diff
@@ -122,32 +118,44 @@ function initialize_common_environment {
...
+            "-v" "${AIRFLOW_SOURCES}:/opt/airflow" \
         )
     elif [[ ${MOUNT_HOST_AIRFLOW_VOLUME} == "true" ]]; then
         print_info
         print_info "Mounting necessary host volumes to Docker"
         print_info
 
         read -r -a EXTRA_DOCKER_FLAGS <<< "$(convert_local_mounts_to_docker_params)"
-        EXTRA_DOCKER_FLAGS+=("-v" "${AIRFLOW_SOURCES}/files:/files" )
+        EXTRA_DOCKER_FLAGS+=( \
```
Comment (feluelle) [inline scripts/ci/libraries/_initialization.sh]: Does the backslash need to be there? Because in line 153 you didn't add it.

**Comment by potiuk** [comment]
> LGTM! But WOW... I didn't know that each build generates 1GB+ of artifacts

Those are recent changes  - they were generated but never published as artifacts. And by looking at those - most of those are Html  coverage reports. - 22 x 45 MB ~ 1GB :).  Thsoe are hardly useful because they are partial coverage reports (only coverage.io when combining them might give us the final, good report. I will remove those in follow-up PR.

**Comment by potiuk** [comment]
Removed here: https://github.com/apache/airflow/pull/10168

File: scripts/ci/libraries/_initialization.sh (hunk 2 above)
Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh]: Indeed. Fixed.

File: scripts/ci/libraries/_initialization.sh [hunk 3]
```diff
@@ -131,7 +131,7 @@ function initialize_common_environment {
         print_info
         print_info "Mount whole airflow source directory for static checks"
         print_info
-        EXTRA_DOCKER_FLAGS=( \
+        EXTRA_DOCKER_FLAGS=(
             "-v" "${AIRFLOW_SOURCES}:/opt/airflow" \
```
Comment (turbaszek) [inline scripts/ci/libraries/_initialization.sh]:
```suggestion
            "-v" "${AIRFLOW_SOURCES}:/opt/airflow"
```
Also redundant? (no backslash in L:144)

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh]: Yep

**Pull request merged**